pkg/disks/loop.go

```diff
@@ -0,0 +1,245 @@
+	}
+
+	symlinkCreated, err := makeDeviceSymlink(loopDevicePath, symlinkPath)
+	if err != nil {
+		return false, fmt.Errorf("can't make symlink to loop device: %w", err)
+	}
+
+	changed := imageCreated || deviceCreated || symlinkCreated
+	return changed, nil
+}
+
+func makeLoopDeviceImage(imagePath string, sizeBytes int64) (bool, error) {
+	imgStat, err := os.Stat(imagePath)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return false, fmt.Errorf("can't stat %q: %w", imagePath, err)
+	}
+	if err == nil {
+		if sizeBytes == imgStat.Size() {
+			klog.V(4).InfoS("Loop device image already exists, nothing to do.", "Image", imagePath)
+			return false, nil
+		}
+		return false, fmt.Errorf("device image resize is not supported, existing image file %q has size %d, different than expected %d", imagePath, imgStat.Size(), sizeBytes)
+	}
+
+	klog.V(4).InfoS("Creating loop device image", "Image", imagePath, "Size", sizeBytes)
+	err = createSparseFile(imagePath, sizeBytes)
+	if err != nil {
+		return false, fmt.Errorf("can't create image file at %q: %w", imagePath, err)
+	}
+
+	return true, nil
+}
+
+func createSparseFile(filePath string, sizeBytes int64) error {
+	f, err := os.Create(filePath)
+	if err != nil {
+		return fmt.Errorf("can't create file at %q: %w", filePath, err)
+	}
+
+	err = f.Close()
+	if err != nil {
+		return fmt.Errorf("can't close file at %q: %w", filePath, err)
+	}
+
+	err = os.Truncate(filePath, sizeBytes)
+	if err != nil {
+		return fmt.Errorf("can't truncate file at %q: %w", filePath, err)
+	}
+
+	return nil
+}
+
+func attachLoopDevice(ctx context.Context, executor exec.Interface, imagePath string) (bool, string, error) {
+	loopDevices, err := losetup.ListLoopDevices(ctx, executor)
+	if err != nil {
+		return false, "", fmt.Errorf("can't list loop devices: %w", err)
+	}
+
+	for _, ld := range loopDevices {
+		if imagePath == ld.BackingFile {
+			klog.V(4).InfoS("Loop device already exists, nothing to do.", "Device", ld.Name, "Image", ld.BackingFile)
+			return false, ld.Name, nil
+		}
+	}
+
+	klog.V(4).InfoS("Attaching loop device to image.", "Image", imagePath)
+	devicePath, err := losetup.AttachLoopDevice(ctx, executor, imagePath)
+	if err != nil {
+		return false, "", fmt.Errorf("can't attach loop device to image %q: %w", imagePath, err)
+	}
+	klog.V(4).InfoS("Successfully attached loop device to image.", "Device", devicePath, "Image", imagePath)
+
+	return true, devicePath, nil
+}
+
+func makeDeviceSymlink(loopDevicePath, devicePath string) (bool, error) {
+	symlinkDirPath := path.Dir(devicePath)
+	_, err := os.Stat(symlinkDirPath)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return false, fmt.Errorf("can't stat %q: %w", symlinkDirPath, err)
+	}
+
+	if errors.Is(err, fs.ErrNotExist) {
+		mkdirErr := os.Mkdir(symlinkDirPath, 0770)
+		if mkdirErr != nil {
+			return false, fmt.Errorf("can't mkdir %q: %w", symlinkDirPath, mkdirErr)
+		}
+	}
```

Comment:
I see you've updated one of the other ones with this (and the one bellow got left. I think a unified flow is better to reason about then checking the same condition at two independent places.
```suggestion
	if err != nil {
		if !errors.Is(err, fs.ErrNotExist) {
			return false, fmt.Errorf("can't stat %q: %w", symlinkDirPath, err)
		}

		mkdirErr := os.Mkdir(symlinkDirPath, 0770)
		if mkdirErr != nil {
			return false, fmt.Errorf("can't mkdir %q: %w", symlinkDirPath, mkdirErr)
		}
	}
```

Reply:
Changed this one and below ones.